chat-commands.js

```diff
@@ -1902,6 +1902,51 @@ const commands = {
 			this.errorReply(`User '${target}' is not locked.`);
 		}
 	},
+	unlockname: function (target, room, user) {
+		if (!target) return this.parse('/help unlock');
+		if (!this.can('lock')) return false;
+
+		const userid = toId(target);
+		const punishment = Punishments.userids.get(userid);
+		if (!punishment) return this.errorReply("This name isn't locked");
+		if (punishment[1] === userid) return this.parse(`/unlock ${userid}`);
+
+		Punishments.userids.delete(userid);
+		Punishments.savePunishments();
+
+		Users.findUsers([userid], []).forEach(curUser => {
+			if (curUser.locked && !curUser.locked.startsWith('#') && Punishments.getPunishType(curUser.userid) !== 'LOCK') {
```

Comment:
.......

I was assuming this would be something like `curUser.locked === userid`.

Any reason why it isn't?

Reply:
Let me put it this way.

Guy is logged into "Spammer" with IP `192.168.0.1` and spams "FUCK MODS" in Lobby.

We use `/lock Spammer`, and now `Spammer` and `192.168.0.1` are locked.

Guy logs out of "Spammer" and into "Spammer 2".

Someone uses `/unlockname Innocent User`.

"Spammer 2" gets unlocked because his username isn't in the punishments table.

Reply:
Oh, wait, `getPunishType` does search IPs.

Still, though. There are probably plenty of more obscure situations this could affect. I don't like the idea of unlocks unlocking unrelated users.